tests/functional/test_install.py

```diff
@@ -37,6 +37,30 @@ def test_pep518_uses_build_env(script, data, common_wheels, command, variant):
     )
 
 
+def test_pep518_refuses_missing_requires(script, data, common_wheels):
+    result = script.pip(
+        'install',
+        '-f', common_wheels,
+        '-f', data.find_links,
+        data.src.join("pep518_missing_requires"),
+        expect_error=True
+    )
+    assert result.returncode == 1
+    assert "does not comply with PEP 518" in result.stderr
+
+
+def test_pep518_refuses_invalid_requires(script, data, common_wheels):
+    result = script.pip(
+        'install',
+        '-f', common_wheels,
+        '-f', data.find_links,
```

Comment:
Same as above.

Reply:
I'd factorize the 2 tests with `@pytest.mark.parametrize`.

Reply:
Nice idea about factorizing the tests.